### ISSUE TITLE
FNX3-22862 ⁃ FennecMigratorException$MigrateAddonsException: java.lang.Exception: Unexpected error: TypeError: extension.setEmbedderDisabled is not a function

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#7875]
https://sentry.prod.mozaws.net/share/issue/edba136dfc4a4c96a2d66032cf93b218/

```
java.lang.Exception: Unexpected error: TypeError: extension.setEmbedderDisabled is not a function
    at org.mozilla.geckoview.WebExtensionController$WebExtensionResult.sendError(WebExtensionController.java:6)
    at org.mozilla.gecko.EventDispatcher$JavaCallbackDelegate$1.run(EventDispatcher.java:3)
    at android.os.Handler.handleCallback(Handler.java:761)
    at android.os.Handler.dispatchMessage(Handler.java:98)
    at android.os.Looper.loop(Looper.java:156)
    at android.app.ActivityThread.main(ActivityThread.java:6523)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:942)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:832)

mozilla.components.support.migration.FennecMigratorException$MigrateAddonsException: java.lang.Exception: Unexpected error: TypeError: extension.setEmbedderDisabled is not a function
    at mozilla.components.support.migration.FennecMigrator.migrateAddons(FennecMigrator.kt:24)
    at mozilla.components.support.migration.FennecMigrator$migrateAddons$1.invokeSuspend(FennecMigrator.kt)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:20)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
    at java.lang.Thread.run(Thread.java:776)
```

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FNX3-22862)

**Comment by csadilek** [comment]
Filed in GV: https://bugzilla.mozilla.org/show_bug.cgi?id=1655761

**Comment by csadilek** [comment]
The most likely cause of this is an ID collision of an installed add-on, and an extension we register dynamically on startup. The only one we have that could cause this is webcompat. Let's revert the following commit on the 48 branch only and see if the problem goes away: https://github.com/mozilla-mobile/android-components/pull/7360/ (We haven't seen it during Nightly and earlier beta migrations before this fix).

This commit fixed the add-on ID of webcompat, now possibly colliding with the installed one in Fennec, which we'd disable anyway during migration, but fail to because of the collision when we query GV.

The revert is NOT needed on newer branches, because we switched to `installBuiltIn` which should not have the problem according to the add-ons team.

**Comment by csadilek** [comment]
We've applied the revert/fix on the 48 branch and should see the numbers taper off once released. As far as we understand, users should not be affected by this.

The same is true for https://github.com/mozilla-mobile/android-components/issues/7878

**Comment by csadilek** [comment]
This is fixed in 48.0.12 (Fenix 79.0.2) and we no longer see this in Sentry. Had to unmerge events though as Sentry combined this with another failure. I will file a follow-up for that other one, which is fixed in GV 80 but we should keep an eye on it.

**Comment by csadilek** [comment]
Marking as Done as this was caused by a race condition without STRs. So, Sentry is our best verification here.
